includes/User.php

```diff
@@ -2167,6 +2167,8 @@ public function setPassword( $str ) {
 				}
 				throw new PasswordError( wfMsgExt( $message, array( 'parsemag' ), $params ) );
 			}
+	            	//Delete all access tokens
+	            	$this->removeAllTokens();
```

Comment:
Are you sure that all invocations of `setPassword()` do need this behaviour (removeAllTokens)?